core/domain/app_feedback_report_domain_test.py

```diff
@@ -400,6 +421,70 @@ def test_get_android_network_type_from_string_invalid_string_raises_error(
                 invalid_network_type)
 
+    def test_from_dict_with_valid_report_dict_returns_expected_type(
+            self) -> None:
+        expected_dict = {
+            'platform_type':
+                app_feedback_report_constants.PLATFORM_CHOICE_ANDROID,
+            'user_supplied_feedback': {
+                'report_type': REPORT_TYPE_SUGGESTION.name,
+                'category': CATEGORY_SUGGESTION_OTHER.name,
+                'user_feedback_selected_items': USER_SELECTED_ITEMS,
+                'user_feedback_other_text_input': USER_TEXT_INPUT
+            },
+            'system_context': {
+                'platform_version': ANDROID_PLATFORM_VERSION,
+                'package_version_code': ANDROID_PACKAGE_VERSION_CODE,
+                'android_device_country_locale_code': COUNTRY_LOCALE_CODE_INDIA,
+                'android_device_language_locale_code':
+                    LANGUAGE_LOCALE_CODE_ENGLISH,
+                'android_device_model': ANDROID_DEVICE_MODEL,
+                'android_sdk_version': ANDROID_SDK_VERSION,
+                'build_fingerprint': ANDROID_BUILD_FINGERPRINT,
+                'network_type': NETWORK_WIFI.value
+            },
+            'device_context': {
+                'platform_version': ANDROID_PLATFORM_VERSION,
+                'package_version_code': ANDROID_PACKAGE_VERSION_CODE,
+                'android_device_country_locale_code': COUNTRY_LOCALE_CODE_INDIA,
+                'android_device_language_locale_code':
+                    LANGUAGE_LOCALE_CODE_ENGLISH,
+                'android_device_model': ANDROID_DEVICE_MODEL,
+                'android_sdk_version': ANDROID_SDK_VERSION,
+                'build_fingerprint': ANDROID_BUILD_FINGERPRINT,
+                'network_type': NETWORK_WIFI.value
+            },
+            'app_context': {
+                'entry_point': {
+                    'entry_point_name': ENTRY_POINT_NAVIGATION_DRAWER
+                },
+                'text_language_code': LANGUAGE_LOCALE_CODE_ENGLISH,
+                'audio_language_code': LANGUAGE_LOCALE_CODE_ENGLISH,
+                'text_size': ANDROID_TEXT_SIZE.value,
+                'only_allows_wifi_download_and_update': True,
+                'automatically_update_topics': False,
+                'account_is_profile_admin': False,
+                'event_logs': EVENT_LOGS,
+                'logcat_logs': LOGCAT_LOGS
+            },
+            'report_submission_timestamp_sec': 0,
+            'android_report_info_schema_version':
+                ANDROID_REPORT_INFO_SCHEMA_VERSION,
+            'report_submission_utc_offset_hrs': 0
+            }
+        revision_obj = self.android_report_obj.from_dict(expected_dict)
+        self.assertTrue(
+            isinstance(
+                revision_obj, app_feedback_report_domain.AppFeedbackReport))
+
+    def test_from_dict_with_invalid_report_dict_raises_error(self) -> None:
+        expected_dict = {
```

Comment:
Is this actually the "expected" dict? It isn't being compared against anything.

Reply:
Changed to 'type_error_dict' as it makes the 'platform_type' = None that raises an error message.